### PR TITLE
Proof of concept for replacing use of spawn with callbacks

### PR DESCRIPTION
**Pull request by spookydonut**
Works with simple testing.

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 1]
```diff
@@ -36,20 +36,25 @@
...
 
+/mob/living/carbon/Xenomorph/Hunter/reset_pounce_delay()
+	. = ..()
+	playsound(src, 'sound/effects/xeno_newlarva.ogg', 50, 0, 1)
+
+/mob/living/carbon/Xenomorph/proc/reset_flags_pass()
+	if(!xeno_caste.hardcore)
+		flags_pass = initial(flags_pass) //Reset the passtable.
+	else
+		flags_pass = 0 //Reset the passtable.
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: Don't we have a NONE define to show its a bit field? If so we might wanna add one for clarity

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 2]
```diff
@@ -87,25 +92,22 @@
...
-		else
-			flags_pass = 0 //Reset the passtable.
-
-	spawn(xeno_caste.pounce_delay)
-		usedPounce = 0
-		to_chat(src, "<span class='xenowarning'><b>You are ready to pounce again.</b></span>")
-		playsound(src, 'sound/effects/xeno_newlarva.ogg', 50, 0, 1)
-		update_action_button_icons()
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/proc/reset_flags_pass), 6)
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/Hunter/reset_pounce_delay), xeno_caste.pounce_delay)
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: No need to include path if you're calling a proc on src

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 3]
```diff
@@ -87,25 +92,22 @@
...
-			flags_pass = initial(flags_pass) //Reset the passtable.
-		else
-			flags_pass = 0 //Reset the passtable.
-
-	spawn(xeno_caste.pounce_delay)
-		usedPounce = 0
-		to_chat(src, "<span class='xenowarning'><b>You are ready to pounce again.</b></span>")
-		playsound(src, 'sound/effects/xeno_newlarva.ogg', 50, 0, 1)
-		update_action_button_icons()
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/proc/reset_flags_pass), 6)
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: Read other comment

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 4]
```diff
@@ -36,20 +36,25 @@
...
-			flags_pass = initial(flags_pass) //Reset the passtable.
-		else
-			flags_pass = 0 //Reset the passtable.
-
-	spawn(xeno_caste.pounce_delay)
-		usedPounce = 0
-		to_chat(src, "<span class='notice'>You get ready to pounce again.</span>")
-		update_action_button_icons()
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/proc/reset_flags_pass), 6)
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/proc/reset_pounce_delay), xeno_caste.pounce_delay)
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: No path needed on src

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 5]
```diff
@@ -36,20 +36,25 @@
...
-		if(!xeno_caste.hardcore)
-			flags_pass = initial(flags_pass) //Reset the passtable.
-		else
-			flags_pass = 0 //Reset the passtable.
-
-	spawn(xeno_caste.pounce_delay)
-		usedPounce = 0
-		to_chat(src, "<span class='notice'>You get ready to pounce again.</span>")
-		update_action_button_icons()
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/proc/reset_flags_pass), 6)
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: No path needed on src

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 6]
```diff
@@ -87,25 +92,22 @@
...
-		playsound(src, 'sound/effects/xeno_newlarva.ogg', 50, 0, 1)
-		update_action_button_icons()
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/proc/reset_flags_pass), 6)
+	addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/Hunter/reset_pounce_delay), xeno_caste.pounce_delay)
 
 	if(stealth && can_sneak_attack) //If we're stealthed and could sneak attack, add a cooldown to sneak attack
 		to_chat(src, "<span class='xenodanger'>Your pounce has left you off-balance; you'll need to wait [HUNTER_POUNCE_SNEAKATTACK_DELAY*0.1] seconds before you can Sneak Attack again.</span>")
 		can_sneak_attack = FALSE
-		sneak_attack_cooldown()
+		addtimer(CALLBACK(src, /mob/living/carbon/Xenomorph/Hunter/proc/sneak_attack_cooldown), HUNTER_POUNCE_SNEAKATTACK_DELAY)
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: Just do it for all timers haha

File: code/modules/mob/living/carbon/xenomorph/Powers.dm [hunk 7]
```diff
@@ -1954,6 +1944,7 @@
 
 	second_wind_delay = world.time + cooldown
 
+	// since this is always 60 seconds or more, do not turn this into a callback.
```
Comment (Qustinnus) [inline code/modules/mob/living/carbon/xenomorph/Powers.dm]: I personally prefer just not having any spawns but @ninjanomnom should look at this first and decide probably

File: code/modules/mob/living/carbon/xenomorph/Powers.dm (hunk 5 above)
Comment (Qustinnus) [reply code/modules/mob/living/carbon/xenomorph/Powers.dm]: Also, why 2 timers if it was originally 1 spawn and why does one have a magic number delay

File: code/modules/mob/living/carbon/xenomorph/Powers.dm (hunk 7 above)
Comment (spookydonut) [reply code/modules/mob/living/carbon/xenomorph/Powers.dm]: arent they both O(logn) for 60 seconds or more?

File: code/modules/mob/living/carbon/xenomorph/Powers.dm (hunk 5 above)
Comment (spookydonut) [reply code/modules/mob/living/carbon/xenomorph/Powers.dm]: it was originally two, from what i understand the spawn(6) reset_flags() is to make it possible to pass through window frames/tables etc when pouncing/thrown but not having the effect last for the whole of the pounce cooldown duration

**Pull request merged**